src/rpc/request.h

```diff
@@ -34,8 +38,9 @@ class JSONRPCRequest
     std::string URI;
     std::string authUser;
     std::string peerAddr;
+    const util::Ref& context;
 
-    JSONRPCRequest() : id(NullUniValue), params(NullUniValue), fHelp(false) {}
+    JSONRPCRequest(const util::Ref& context) : id(NullUniValue), params(NullUniValue), fHelp(false), context(context) {}
```

Comment:
`Ref& _context` and `context(_context)`? We usually use different names for arguments and members, don't we?

Reply:
This will cause -Wshadow warnings, but luckily we don't have them enabled anymore

Reply:
This can be fixed up in the follow-up that switches to C++17